internal/cmd/context/command_delete.go

```diff
@@ -27,20 +23,31 @@ func (c *command) newDeleteCommand() *cobra.Command {
 }
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
-	ctx, err := c.Config.FindContext(args[0])
-	if err != nil {
+	if confirm, err := c.confirmDeletion(cmd, args); err != nil {
 		return err
+	} else if !confirm {
+		return nil
 	}
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.Context, ctx.Name)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
-		return err
+	deleteFunc := func(id string) error {
+		return c.Config.DeleteContext(id)
 	}
 
-	if err := c.Config.DeleteContext(ctx.Name); err != nil {
+	deleted, err := resource.Delete(args, deleteFunc, nil)
+	resource.PrintDeleteSuccessMsg(deleted, resource.Context)
+
+	return err
+}
+
+func (c *command) confirmDeletion(cmd *cobra.Command, args []string) (bool, error) {
+	describeFunc := func(id string) error {
+		_, err := c.Config.FindContext(id)
 		return err
 	}
 
-	output.Printf(errors.DeletedResourceMsg, resource.Context, ctx.Name)
-	return nil
+	if err := resource.ValidateArgs(pcmd.FullParentName(cmd), args, resource.Context, describeFunc); err != nil {
```

Comment:
Let's just pass `cmd` directly? We're calling `pcmd.FullParentName()` 24 times

Reply:
I tried this and remembered why I did it this way: I get an import cycle error because the `cmd` pkg depends on `resource`. I think the `cmd` package is probably too large, which makes import cycles easy to run into.

I guess the easy thing to do is to just move `FullParentName` to `resource`, but it's called in one other place and `cmd` feels like the natural place for this function.

Does that seem like an alright solution for now?